src/utils.ts

```diff
@@ -216,27 +216,30 @@ const utils = {
   /**
    * Secure compare, from https://github.com/freewil/scmp
    */
-  secureCompare: (a: Uint8Array, b: Uint8Array): boolean => {
-    a = Buffer.from(a);
-    b = Buffer.from(b);
+  secureCompare: (a: string, b: string): boolean => {
+    if (!a || !b) {
+      throw new Error('secureCompare must receive two arguments');
+    }
+    const aEncoded: Uint8Array = new TextEncoder().encode(a);
+    const bEncoded: Uint8Array = new TextEncoder().encode(b);
 
     // return early here if buffer lengths are not equal since timingSafeEqual
     // will throw if buffer lengths are not equal
-    if (a.length !== b.length) {
+    if (aEncoded.length !== bEncoded.length) {
       return false;
     }
 
     // use crypto.timingSafeEqual if available (since Node.js v6.6.0),
     // otherwise use our own scmp-internal function.
     if (crypto.timingSafeEqual) {
-      return crypto.timingSafeEqual(a, b);
+      return crypto.timingSafeEqual(aEncoded, bEncoded);
```

Comment:
how did this call work before? does it allow strings?

Reply:
`crypto.timingSafeEqual` doesn't allow strings. Before we would convert the arguments to Buffers like `a = Buffer.from(a)`. Removing these conversions makes throw the error `TypeError: The "buf1" argument must be an instance of Buffer, TypedArray, or DataView.`

Reply:
My bad, missed those lines. Let's move 
```
    const aEncoded: Uint8Array = new TextEncoder().encode(a);
    const bEncoded: Uint8Array = new TextEncoder().encode(b);
```
calls inside this if statement. We can rewrite the loop to operate on chars using `a.charCodeAt() ^ b. charCodeAt()`.